controllers/cluster_upgrade.go

```diff
@@ -239,10 +239,10 @@ func IsPodNeedingRollout(status postgres.PostgresqlStatus, cluster *apiv1.Cluste
 	}
 
 	if configuration.Current.EnableAzurePVCUpdates {
-		for _, pvc := range cluster.Status.ResizingPVC {
-			// This code works on the assumption that the PVC have the same name as the pod using it.
-			if status.Pod.Name == pvc {
-				return true, false, fmt.Sprintf("rebooting pod to complete resizing %s", pvc)
+		for _, pvcName := range cluster.Status.ResizingPVC {
+			// This code works on the assumption that the PVC beings with the name of the pod using it.
+			if specs.DoesBelongToInstance(status.Pod.Name, pvcName) {
```

Comment:
We should verify here that we are rebooting the instance just one time after we have resized all the PVC that need resizing for a certain instance.

Reply:
Talking with @armru we decided that in this moment we don't have enough information to correctly implement this. The worst case here is when the user changes the storage configuration of the pgdata and the storage configuration of the WALs at the same time. In that case the operator may restart that Pod twice.

Let's address this in a separate ticket as this behavior may be desired or not depending on the situation.

@armru can you create a ticket for that?